### PR TITLE
fix: status error in purchase invoice

### LINKED ISSUE #21932: Click on "unpaid" purchase invoice in list view causes "invalid filter" error
## Description of the issue
Clicking on the status field on an "unpaid" purchase invoice in the purchase invoice list leads to the error "Invalid filter: "due""

## Context information (for bug reports)

**Output of `bench version`**
```
erpnext 12.8.0
frappe 12.5.2
```

## Steps to reproduce the issue

1. Click on an unpaid invoice on the "status" field column (unpaid) on the accounting list "purchase invoice".

### Observed result
The error popup "Invalid filter: "due"" occurs.

### Expected result
The list will be filtered accordingly.

## Additional information
The error is probably in the file "erpnext/accounts/doctype/purchase_invoice/purchase_invoice_list.js" where the field name "due" is used. It should be "due_date" instead.

### PR DESCRIPTION
**Pull request by kennethsequeira**
Fixes: https://github.com/frappe/erpnext/issues/21932

Steps:
1. Go to purchase invoice
2. Click on Unpaid status in list view (not filter)
3. Error

Before fix:

purchase_invoice_unpaid

After fix:

purchase_invoice_fix

### CODE REVIEW

**Comment by kennethsequeira** [comment]
@Mergifyio backport version-12-hotfix

**Comment by kennethsequeira** [comment]
@Mergifyio version-11-hotfix

**Comment by kennethsequeira** [comment]
@Mergifyio backport version-11-hotfix

**Review by marination — DISMISSED** [review]

File: erpnext/accounts/doctype/purchase_invoice/purchase_invoice_list.js [hunk 1]
```diff
@@ -25,3 +25,4 @@ frappe.listview_settings['Purchase Invoice'] = {
 		}
 	}
 };
+s
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/purchase_invoice/purchase_invoice_list.js]: Codacy is failing because of this

**Comment by kennethsequeira** [comment]
@deepeshgarg007 fixed the codacy issue. Kindly re-review it

**Review by deepeshgarg007 — APPROVED** [review]

**Pull request merged**